src/App.tsx

```diff
@@ -1,61 +1,103 @@
+import React, { useState } from 'react';
 import './App.scss';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errors, setErrors] = useState({ title: '', userId: '' });
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newErrors = { title: '', userId: '' };
+    let hasError = false;
+
+    if (!title.trim()) {
+      newErrors.title = 'Please enter a title';
+      hasError = true;
+    }
+
+    if (!userId) {
+      newErrors.userId = 'Please choose a user';
+      hasError = true;
+    }
+
+    if (hasError) {
+      setErrors(newErrors);
+      return;
+    }
+
+    const newTodo = {
+      id: todos.length ? Math.max(...todos.map(todo => todo.id)) + 1 : 1,
+      title: title.trim(),
+      userId,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+    setTitle('');
+    setUserId(0);
+    setErrors({ title: '', userId: '' });
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.replace(/[^a-zA-Z0-9\s]/g, '');
```

Comment:
The regex `/[^a-zA-Z0-9\s]/g` restricts the title to only alphanumeric characters and spaces. If the task description allows for more characters (e.g., punctuation), consider adjusting this regex.